### PR TITLE
AtomicInterval uses more strict ordering

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: sdk/src/timing.rs [hunk 1]
```diff
@@ -81,19 +81,19 @@ impl AtomicInterval {
     /// except, if skip_first=false, false until 'interval_time_ms' has elapsed since this struct was created
     pub fn should_update_ext(&self, interval_time_ms: u64, skip_first: bool) -> bool {
         let now = timestamp();
-        let last = self.last_update.load(Ordering::Relaxed);
+        let last = self.last_update.load(Ordering::Acquire);
         now.saturating_sub(last) > interval_time_ms
             && self
                 .last_update
-                .compare_exchange(last, now, Ordering::Relaxed, Ordering::Relaxed)
+                .compare_exchange(last, now, Ordering::AcqRel, Ordering::Acquire)
```
Comment (brooksprumo) [inline sdk/src/timing.rs:88]: I think the second ordering can stay `Relaxed` since the failure value is not used.

File: sdk/src/timing.rs [hunk 2]
```diff
@@ -81,19 +81,19 @@ impl AtomicInterval {
...
                 == Ok(last)
             && !(skip_first && last == 0)
```
Comment (brooksprumo) [inline sdk/src/timing.rs:90]: I wonder why this line is _after_ the `.compare_exchange()`?

**Review by brooksprumo — COMMENTED** [review]
There's a few other spots in the codebase that use `AtomicInterval`. Most are for stats/logging, which I think will be fine with this change. The rest look like enforcing some timeout-like interval, which I think should also be fine. The ones behind an `Arc` likely should have this fix to make them multi-thread safe.

But, do you know if this change fixes the bug you're looking at? And/or have you been able to see if any perf numbers change? I don't want to contribute to slow blocks 😅

**Comment by sakridge** [comment]
Mostly it's just used for metrics right?

**Comment by jeffwashington** [comment]
does not seem necessary atm

**Pull request closed without merging**